packages/product-json-to-csv/test/map-product-data.spec.js

```diff
@@ -0,0 +1,622 @@
+        id: '12345ab-id',
+        key: 'product-key',
+        productType: 'resolved-product-type',
+        variant: {
+          id: 1,
+          sku: 'A0E200000001YKI',
+        },
+        article: 'sample 089 WHT',
+        designer: 'michaelkors',
+        color: 'white',
+        colorFreeDefinition: {
+          en: 'black-white',
+          de: 'schwarz-weiß',
+        },
+        addedAttr: '',
+        anotherAddedAttr: '',
+        state: 'my-resolved-state',
+        hasStagedChanges: false,
+      }
+      expect(productMapping._mapProperties(sample)).toEqual(expected)
+    })
+
+    it('converts variant image array to strings', () => {
+      const sample = {
+        id: '12345ab-id',
+        key: 'product-key',
+        variant: {
+          id: 1,
+          sku: 'A0E200000001YKI',
+          images: [
+            {
+              url: 'https://example.com/foobar/commer.jpg',
+              dimensions: {
+                w: 3,
+                h: 4,
+              },
+            },
+            {
+              url: 'https://example-2.com/demo/tools.jpg',
+              dimensions: {
+                w: 1,
+                h: 5,
+              },
+              label: 'image-label',
+            },
+          ],
+        },
+        state: {
+          key: 'my-resolved-state',
+        },
+        hasStagedChanges: false,
+      }
+      const expected = {
+        id: '12345ab-id',
+        key: 'product-key',
+        variant: {
+          id: 1,
+          sku: 'A0E200000001YKI',
+          images: oneLineTrim`
+            https://example.com/foobar/commer.jpg|3|4;
+            https://example-2.com/demo/tools.jpg|1|5|image-label`,
+        },
+        state: 'my-resolved-state',
+        hasStagedChanges: false,
+      }
+      expect(productMapping._mapProperties(sample)).toEqual(expected)
+    })
+  })
+
+  describe('::mapCategories', () => {
+    it('resolves categories to string by name', () => {
+      const sampleCat = [
+        {
+          name: {
+            en: 'cat-in-en',
+            de: 'cat-in-de',
+          },
+          externalId: 'cat-ext-id',
+        },
+        {
+          name: {
+            en: 'cat-foo-en',
+            de: 'cat-foo-de',
+          },
+          externalId: 'cat-foo-id',
+        },
+      ]
+      const expected = 'cat-in-en;cat-foo-en'
+      const args = ['name', ';', 'en']
+      expect(ProductMapping._mapCategories(sampleCat, ...args)).toBe(expected)
+    })
+
+    it('resolves categories to string by externalId', () => {
+      const sampleCat = [
+        {
+          name: {
+            en: 'cat-in-en',
+            de: 'cat-in-de',
+          },
+          externalId: 'cat-ext-id',
+        },
+        {
+          name: {
+            en: 'cat-foo-en',
+            de: 'cat-foo-de',
+          },
+          externalId: 'cat-foo-id',
+        },
+      ]
+      const expected = 'cat-ext-id;cat-foo-id'
+      const args = ['externalId', ';']
```

Comment:
These are not really args. They are something additional we spread into the function applied. Should we rename these accordingly?